### PR TITLE
feat: Support temporary audio focus request

### PR DESCRIPTION
**Pull request by tremblaymathjive**

### CODE REVIEW

File: android/src/main/java/com/zmxv/RNSound/RNSoundModule.java [hunk 1]
```diff
@@ -289,7 +289,7 @@ public void stop(final Double key, final Callback callback) {
     }
 
     // Release audio focus in Android system
-    if (!this.mixWithOthers && key == this.focusedPlayerKey) {
```
Comment (tremblaymathjive) [inline android/src/main/java/com/zmxv/RNSound/RNSoundModule.java:292]: Comparing 2 doubles always return false 😱

File: android/src/main/java/com/zmxv/RNSound/RNSoundModule.java [hunk 2]
```diff
@@ -226,7 +226,7 @@ public void play(final Double key, final Callback callback) {
     if (!this.mixWithOthers) {
       AudioManager audioManager = (AudioManager) context.getSystemService(Context.AUDIO_SERVICE);
 
-      audioManager.requestAudioFocus(this, AudioManager.STREAM_MUSIC, AudioManager.AUDIOFOCUS_GAIN);
+      audioManager.requestAudioFocus(this, AudioManager.STREAM_MUSIC, AudioManager.AUDIOFOCUS_GAIN_TRANSIENT);
```
Comment (tremblaymathjive) [inline android/src/main/java/com/zmxv/RNSound/RNSoundModule.java:229]: `AUDIOFOCUS_GAIN_TRANSIENT` Used to indicate a temporary gain or request of audio focus, anticipated to last a short amount of time. Examples of temporary changes are the playback of driving directions, or an event notification.

Comment (wilau2) [reply android/src/main/java/com/zmxv/RNSound/RNSoundModule.java:229]: Does it mean we will still hear the music in background while playing the voicemail ?

Comment (tremblaymathjive) [reply android/src/main/java/com/zmxv/RNSound/RNSoundModule.java:229]: No
AUDIOFOCUS_GAIN_TRANSIENT_MAY_DUCK is the option your are referring to
```
    * Used to indicate a temporary request of audio focus, anticipated to last a short
     * amount of time, and where it is acceptable for other audio applications to keep playing
     * after having lowered their output level (also referred to as "ducking").
     * Examples of temporary changes are the playback of driving directions where playback of music
     * in the background is acceptable.
```

**Review by wilau2 — APPROVED** [review]

**Pull request merged**